src/v/iceberg/conversion/tests/iceberg_protobuf_tests.cc

```diff
@@ -544,6 +554,161 @@ TEST(values_protobuf, TestTimestamp) {
       ElementsAre(OptionalIcebergPrimitive<timestamp_value>(1743540027001635)));
 }
 
+TEST(values_protobuf, TestStruct) {
```

Comment:
Many of these tests are quite hard to read with all the manual construction of fields... Is there an easier way to ensure that they are both equal?

Maybe we need something like:

```
EXPECT_THAT(serialized_string, IsJSON(R"({"foo":"bar"})"));
```

And we can define `IsJSON` to be a `MATCHER_P`: https://google.github.io/googletest/reference/matchers.html#defining-matchers

Which deserializes the json using rapid json and then does equality checks on the DOM. I think we have code for this in other places.